base/compiler/abstractinterpretation.jl

```diff
@@ -1679,15 +1679,18 @@ function abstract_invoke(interp::AbstractInterpreter, (; fargs, argtypes)::ArgIn
     ft = widenconst(ft′)
     ft === Bottom && return CallMeta(Bottom, EFFECTS_THROWS, NoCallInfo())
     (types, isexact, isconcrete, istype) = instanceof_tfunc(argtype_by_index(argtypes, 3))
-    types === Bottom && return CallMeta(Bottom, EFFECTS_THROWS, NoCallInfo())
+    unwrapped = unwrap_unionall(types)
+    if types === Bottom || types === Any || !(unwrapped isa DataType)
+        return CallMeta(Bottom, EFFECTS_THROWS, NoCallInfo())
```

Comment:
This doesn't seem right. If types is `Any`, we don't know much about what the types are, so it seems hard to believe that we would be able to prove that it throws. Perhaps you'd want to add these conditions to the `isexact` check on the next line instead?

Reply:
we know that it is exactly a literal `Any` here (from the previous check), not `<:Any`

Reply:
ah, nevermind, I got the order of those tests mixed up, and we are indeed now bypassing the `isexact` check

Reply:
Isn't that only true if `isexact` is set?

Reply:
The `isexact` check can also just be moved above this, since `types===Union{}` will always also be `isexact`